lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -368,13 +370,13 @@ struct StoreOpConversion
           assert(elemOffset < valueElems.size());
           Value elem = valueElems[elemOffset];
           if (elem.getType().isInteger(1))
-            elem = rewriter.create<LLVM::SExtOp>(loc, type::i8Ty(ctx), elem);
+            elem = sext(i8_ty, elem);
           elem = bitcast(elem, valueElemTy);
           llWord = insert_element(wordTy, llWord, elem, i32_val(elemIdx));
         }
         llWord = bitcast(llWord, valArgTy);
 #ifdef USE_ROCM
-        Value maskVal = llMask ? maskElems[vecStart] : int_val(1, 1);
+        Value maskVal = llMask ? and_(mask, maskElems[vecStart]) : mask;
```

Comment:
I found this change the culprit of the failure of test_convert2d and this is not from upstream so we cannot find anything suspicious in [that PR#1322](https://github.com/openai/triton/commit/9b7c65a3a9639bb91bc5b7ce3017bf33c8347dd5). I'll create a PR to further explain the issue.